### PR TITLE
DEVPROD-4090 Error when receiving a server error when validating config

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-4090

### Description
Before, during config validation we would just no-op when we get an error. This treats the error as an error and says it to the user.

### Testing
Built and ran locally. I didn't add unit testing since it's not that easy to add for this one, and pretty niche. We don't test that in our other commands from what I saw- since the crux of it is just API requests that error for this command return nil rather than the error (seems like a typo / mishap, git blame dates it back 6 years).

### CODE REVIEW

File: operations/validate.go [hunk 1]
```diff
@@ -50,20 +50,12 @@ func Validate() cli.Command {
...
 			}
-			ctx, cancel := context.WithCancel(context.Background())
-			defer cancel()
 
 			conf, err := NewClientSettings(confPath)
 			if err != nil {
 				return errors.Wrap(err, "loading configuration")
 			}
 
-			client, err := conf.setupRestCommunicator(ctx, !quiet)
```
Comment (ZackarySantana) [inline operations/validate.go:61]: code clean up- wasn't used anywhere and same with the context once you remove this.

Comment (Kimchelly) [reply operations/validate.go:61]: Probably someone once having the small hope that we would one day get rid of the legacy REST API and use REST v2 🤷

**Review by Kimchelly — APPROVED** [review]
Seems reasonable to me! I can't think of any reason why validation should be allowed to no-op if the request encountered an error (auth-related or otherwise).

**Comment by ZackarySantana** [comment]
I thought the same that someone had that hope... Feels bad to squash it 😭

**Pull request merged**